src/routes/player/WaitingForNextQuestion.tsx

```diff
@@ -0,0 +1,87 @@
+import { useEffect, useState } from "react";
+import { socket } from "../../socket";
+import { useNavigate, useParams } from "react-router-dom"
+import { observable } from "@legendapp/state";
+
+// global state
+const state$ = observable({
+  player: {
+    id: '',
+  },
+  question: '',
+  answers: [],
+  timer: 30,
+
+});
+
+function WaitingForNextQuestion() {
+
+    const navigate = useNavigate();
+    const { room_id: roomId } = useParams();
+    const [waitingForNextQuestion, setWaitingForNextQuestion] = useState<boolean>(true);
+
+    useEffect(() => {
+        // handle newQuestion event to send recvQuestion event to server
+        const handleNewQuestion = (data: { question: string, answers: [] }) => {
+
+            state$.question.set(data.question);
+            state$.answers.set(data.answers);
+
+            // send WS event to server to start timer
+            socket.emit('recvQuestion', roomId);
+        }
+
+        // receive next question
+        socket.on("newQuestion", handleNewQuestion);
+    });
+
+    useEffect(() => {
+        // handle startTimer event to display question
+        const handleStartTimer = () => {
+            // update page
+            setWaitingForNextQuestion(false);
+        }
+        
+        // start timer
+        socket.on("startTimer", handleStartTimer);
+    })
+
+    useEffect(() => {
```

Comment:
since we are using legend for state management, we want to avoid using hooks and stick to using legend. Look through the docs and see if there is an alternative way to do this.